contracts/token/ERC721/ERC721Metadata.sol

```diff
@@ -56,7 +59,16 @@ contract ERC721Metadata is Context, ERC165, ERC721, IERC721Metadata {
      * Throws if the token ID does not exist. May return an empty string.
```

Comment:
We have to change the documentation here to explain that if there is a base URI this function will return the concatenation of it with the "specific" token URI, and also explain the _why_: that using this feature will result in using less gas in the setter.

"May return an empty string" is not valid in the same way anymore: the function currently returns the base URI when this token doesn't have its own. @nachomazzara I'm curious to hear from your experience what would be the best behavior in this scenario. My intuition is that we should have a "default" URI that is used, rather than returning the base URI itself.

Reply:
> We have to change the documentation here to explain that if there is a base URI this function will return the concatenation of it with the "specific" token URI, and also explain the why: that using this feature will result in using less gas in the setter.

Nice catch! implemented in the next commit.

> "May return an empty string" is not valid in the same way anymore: the function currently returns the base URI when this token doesn't have its own. @nachomazzara I'm curious to hear from your experience what would be the best behavior in this scenario. My intuition is that we should have a "default" URI that is used, rather than returning the base URI itself.

I thought about it and that is why I keep the same behavior as before: `May return an empty string`
The code is: 

```solidity
function tokenURI(uint256 tokenId) public view returns (string memory) {
        return string(abi.encodePacked(_baseTokenURI(), _tokenURI(tokenId)));
    }
```

Where `_tokenURI` reverts if the  _[...] token ID does not exist_.

My rationale: 
- Keep the ERC721Metadata standard behavior for `tokenURI`.
- If we return a _default_ URI we are defining behavior that can be implemented using a function `baseTokenURI()` as I did at [ERC721FullMock.sol](https://github.com/nachomazzara/openzeppelin-contracts/blob/master/contracts/mocks/ERC721FullMock.sol#L34).  In our current use case, we want to return an empty string/throw If someone wants to get the `tokenURI` for a non-existing id.

This lets the door opens to:
- Returns an empty string/throw which is the standard behavior of the ERC721Metadata `tokenURI` method.
- Override it and return the base in case of the token does not exist.
- Override it and return a default URI.

I would keep it as it is, letting the developers define the behavior they need using the internal method exposed. BUT, if you still think that we should change it, let's do it.